src/lib/server/hyrise_session.cpp

```diff
@@ -23,278 +23,270 @@
-  _socket.async_read_some(
-      boost::asio::buffer(_input_packet.data, size),
-      boost::bind(next_handler, this, boost::asio::placeholders::error, boost::asio::placeholders::bytes_transferred));
-}
-
-void HyriseSession::_send_ssl_denied() {
-  // Don't use new_output_packet here, because this packet has special size requirements (only contains N, no size)
-  OutputPacket output_packet;
-  PostgresWireHandler::write_value(output_packet, NetworkMessageType::SslNo);
-  async_send_packet(output_packet);
-  _async_flush();
-
-  _async_receive_header(STARTUP_HEADER_LENGTH);
-}
-
-void HyriseSession::_send_auth() {
-  // This packet is our AuthenticationOK, which means we do not require any auth.
-  OutputPacket output_packet = PostgresWireHandler::new_output_packet(NetworkMessageType::AuthenticationRequest);
-  PostgresWireHandler::write_value(output_packet, htonl(0u));
-  async_send_packet(output_packet);
-
-  _send_ready_for_query();
-}
-
-void HyriseSession::_send_ready_for_query() {
-  _state = SessionState::WaitingForQuery;
-  // ReadyForQuery packet 'Z' with transaction status Idle 'I'
-  OutputPacket output_packet = PostgresWireHandler::new_output_packet(NetworkMessageType::ReadyForQuery);
-  PostgresWireHandler::write_value(output_packet, TransactionStatusIndicator::Idle);
-  async_send_packet(output_packet);
-  _async_flush();
-
-  // Now we wait for the next query to come
-  _async_receive_header();
-}
-
-void HyriseSession::_accept_query() {
-  _state = SessionState::SimpleQuery;
-  const auto sql = PostgresWireHandler::handle_query_packet(_input_packet, _expected_input_packet_length);
-  const std::vector<std::shared_ptr<ServerTask>> tasks = {std::make_shared<CreatePipelineTask>(_self, sql)};
-  CurrentScheduler::schedule_tasks(tasks);
+boost::future<void> HyriseSession::_handle_simple_query_command(const std::string sql) {
+  auto create_sql_pipeline = [=] () {
+    return _dispatch_server_task(std::make_shared<CreatePipelineTask>(sql));
+  };
+  
+  auto load_table_file = [=](std::string& file_name, std::string& table_name) {
+    auto task = std::make_shared<LoadServerFileTask>(file_name, table_name);
+    return _dispatch_server_task(task)
+      >> then >> [=] () { return _connection->send_notice("Successfully loaded " + table_name); };
+  };
+  
+  auto execute_sql_pipeline = [=] (std::shared_ptr<SQLPipeline> sql_pipeline) {
+    auto task = std::make_shared<ExecuteServerQueryTask>(sql_pipeline);
+    return _dispatch_server_task(task)
+      >> then >> [=] () { return sql_pipeline; };
+  };
+  
+  auto send_query_response = [=](std::shared_ptr<SQLPipeline> sql_pipeline) {
+    auto task = std::make_shared<SendQueryResponseTask>(_connection, sql_pipeline, sql_pipeline->get_result_table());
+    return _dispatch_server_task(task);
+  };
+    
+  return create_sql_pipeline() >> then >> [=] (std::shared_ptr<CreatePipelineResult> result) {
+    return result->is_load_table
```

Comment:
let's use an explicit `if () {} else {}` here, the ternary if with the >> then is confusing